src/js/modifiers.js

```diff
@@ -433,17 +434,19 @@ function modify_update_approximations() {
     if (maxcentsd < 0)
       maxcentsd = 0;
 
-    var menulength = (semiconvergents) ? current_approximations.ratios.length : current_approximations.convergent_indicies.length;
+    var menulength = (semiconvergents) ? currentRatioStructure.length : currentRatioStructure.cf.length;
```

Comment:
why do we need the parenthesis around semiconvergents?

Reply:
Will clean this whole function up when I update it to use the model